File: 프로그래머스/3/43162. 네트워크/네트워크.py


def solution(n, computers):
    answer = 0
    
    def dfs(cur, bef):
        for nxt in range(n):
            if nxt == bef:
                continue
            if computers[cur][nxt] and not v[nxt]:
                v[nxt] = 1
                dfs(nxt, cur)
    
    v = [0] * n
    for i in range(n):
        if not v[i]:
            v[i] = 1
            dfs(i, -1)
            answer += 1
    
    return answer

File: 프로그래머스/3/43162. 네트워크/네트워크.js
function solution(n, computers) {
    let answer = 0;
    
    // dfs
    function dfs(cur) {
        for (let nxt=0; nxt<n; ++nxt) {
            if (!V[nxt] && (computers[cur][nxt] === 1)) {
                V[nxt] = true;
                dfs(nxt);                
            }
        }
    }
    
    const V = Array.from({length: n}, ()=> false);
    for (let i=0; i<n; ++i) {
        if (!V[i]) {
            V[i] = true;
            dfs(i, -1);
            answer++;
        }
    }
    return answer;
}